code/States/MapSelectionState.cs

```diff
@@ -69,4 +49,50 @@ public static void SetVote( string map )
 
 		state.Votes[player.Client] = map;
 	}
+
+	private static async Task<List<string>> GetLocalMapIdents()
+	{
+		var maps = new List<string>();
+
+		var rawMaps = FileSystem.Data.ReadAllText( "maps.txt" );
+		if ( rawMaps.IsNullOrEmpty() )
+			return maps;
+
+		var splitMaps = rawMaps.Split( "\n" );
+		foreach ( var rawMap in splitMaps )
+		{
+			var mapIdent = rawMap.Trim();
+			var package = await Package.Fetch( mapIdent, true );
+			if ( package is not null && package.PackageType == Package.Type.Map )
```

Comment:
@kolexxx should we do a check serverside to see if the map is valid? or just let the server crash due to the inability of the server host, we can discuss tomorrow.